### PR TITLE
Suppressing warnings for malformed metadata in product template downloads

### LINKED ISSUE #32137: PHP Warning when adding a Digital product from template
### Describe the bug

The following PHP warning is visible when adding a `Digital product` from a template.

```
exif_read_data(album-1.jpg): Incorrect APP1 Exif Identifier Code
 Type: PHP Warning Line: 829
 File: /wp-admin/includes/image.php
```

### To Reproduce

1. Go to the `Home` screen
2. Press `Add my products`
3. And then press `Start with a template`
4. Select `Digital product`
5. The PHP warning below will be visible
 ```
exif_read_data(album-1.jpg): Incorrect APP1 Exif Identifier Code
 Type: PHP Warning Line: 829
 File: /wp-admin/includes/image.php
```

### Screenshots

screenshot-misty-caribou jurassic ninja-2022 03 08-18_15_00

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This resolves a warning that is being triggered when a user creates a product via the digital download template. The warning results from corrupt meta-data on an image file that is downloaded (from [this source](https://woocommercecore.mystagingwebsite.com/wp-content/uploads/2017/12/album-1.jpg)) and then used within the product template as the product image. This doesn't cause any functional issue other than the warning when `WP_DEBUG` mode is set to `true`.

A couple options to resolve this:
1) Fix the EXIF data on the image itself, although this is sourced at `https://woocommercecore.mystagingwebsite.com/wp-content/uploads/2017/12/album-1.jpg`, and I'm unsure how this is edited or accessed.
2) We could consider this just too noisy as it would be triggered from any image with questionable metadata, with the solution being to suppress warnings in this instance.

I've taken option 2 for this PR. 

Closes #32137  .

### How to test the changes in this Pull Request:

1. Checkout branch
2. Ensure `WP_DEBUG` is set to true in `wp-config.php`
3. Go to the Home screen
4. Press Add my products on task list
5. And then press Start with a template
6. Select Digital product
7. Ensure no warning is displayed here or within the `debug.log`

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/wc-rest-functions.php [hunk 1]
```diff
@@ -139,7 +139,7 @@ function wc_rest_set_uploaded_image_as_attachment( $upload, $id = 0 ) {
 		include_once ABSPATH . 'wp-admin/includes/image.php';
 	}
 
-	$image_meta = wp_read_image_metadata( $upload['file'] );
+	$image_meta = @wp_read_image_metadata( $upload['file'] );
```
Comment (joelclimbsthings) [inline plugins/woocommerce/includes/wc-rest-functions.php:142]: Adding the `@` here will suppress any warnings from calling this function. I noticed other calls to `wp_read_image_metadata` within WooCommerce are already doing this.

**Review by joshuatf — COMMENTED** [review]
This looks like a good call, but would like to get this addressed at its root as well.  I'll ping you with details on getting admin access to the site.

**Comment by mattsherman** [comment]
@joelclimbsthings What's the status of this PR?

**Comment by joelclimbsthings** [comment]
> What's the status of this PR?

@mattsherman I had been prioritizing the [task chaining side panel PR](https://github.com/woocommerce/woocommerce/pull/33124), but I was also able to gain access to the `https://woocommercecore.mystagingwebsite.com` to replace this image at the source. Looking at it now.

**Comment by joelclimbsthings** [comment]
@joshuatf Alright, I've updated the image on the staging site, and updated all references to the new image. I also tested on a fresh site an fresh clone of this branch, and the correct image was downloaded without issue. Have at it!

Edit: The diff is a bit hard to read, but the only change(s) regarding the image is the date in the path.

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/32971/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Updating references to old album-1 image` 9e62fb58b8ee0c4bb888262e462cbae6fc7c4fdc
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Review by joshuatf — APPROVED** [review]
Testing well, thanks for the fix, @joelclimbsthings!

**Pull request merged**